bridge/setu/listener/heimdall.go

```diff
@@ -65,15 +65,34 @@ func (hl *HeimdallListener) StartPolling(ctx context.Context, pollInterval time.
 	ticker := time.NewTicker(interval)
 
 	var eventTypes []string
-	eventTypes = append(eventTypes, "message.action='checkpoint'")
-	eventTypes = append(eventTypes, "message.action='event-record'")
+	// eventTypes = append(eventTypes, "message.action='checkpoint'")
+	// eventTypes = append(eventTypes, "message.action='event-record'")
+	// eventTypes = append(eventTypes, "message.action='tick'")
+	// ADD EVENT TYPE for SLASH-LIMIT
 
 	// start listening
 	for {
 		select {
 		case <-ticker.C:
-			fromBlock, toBlock := hl.fetchFromAndToBlock()
-			if fromBlock < toBlock {
+			fromBlock, toBlock, err := hl.fetchFromAndToBlock()
+			if err != nil {
+				hl.Logger.Error("Error fetching fromBlock and toBlock...skipping events query", "error", err)
+			} else if fromBlock < toBlock {
+
+				hl.Logger.Info("Fetching new events between", "fromBlock", fromBlock, "toBlock", toBlock)
+
+				// Querying and processing Begin events
+				for i := fromBlock; i <= toBlock; i++ {
+					events, err := helper.GetBeginBlockEvents(hl.httpClient, int64(i))
+					if err != nil {
+						hl.Logger.Error("Error fetching begin block events", "error", err)
+					}
+					for _, event := range events {
+						hl.ProcessBlockEvent(sdk.StringifyEvent(event), int64(i))
+					}
+				}
+
+				// Querying and processing tx Events. Below for loop is kept for future purpose to process events from tx
 				for _, eventType := range eventTypes {
```

Comment:
Is this loop required?

Reply:
It's kept for future incase we listen to tx events. 

currently it's of no use